src/muse/timeslices.py

```diff
@@ -105,29 +76,191 @@ def setup_module(settings: Union[str, Mapping]):
     TIMESLICE = read_timeslices(settings)
 
 
-def broadcast_timeslice(x, ts=None):
+def broadcast_timeslice(
+    data: DataArray, ts: Optional[DataArray] = None, level: Optional[str] = None
+) -> DataArray:
+    """Convert a non-timesliced array to a timesliced array by broadcasting.
+
+    If x is already timesliced in the appropriate scheme, it will be returned unchanged.
+
+    Args:
+        data: Array to broadcast.
+        ts: Dataarray with timeslice lengths. If None, defaults to the global timeslice.
+        level: Level to broadcast to. If None, use the finest level of ts.
+
+    """
     from xarray import Coordinates
 
     if ts is None:
         ts = TIMESLICE
 
-    # If x already has timeslices, check that it is matches the reference timeslice.
-    if "timeslice" in x.dims:
-        if x.timeslice.reset_coords(drop=True).equals(ts.timeslice):
-            return x
+    if level is not None:
+        ts = compress_timeslice(ts, ts=ts, level=level, operation="sum")
+
+    # If x already has timeslices, check that it matches the reference timeslice.
+    if "timeslice" in data.dims:
+        if data.timeslice.reset_coords(drop=True).equals(ts.timeslice):
+            return data
         raise ValueError("x has incompatible timeslicing.")
```

Comment:
Now it is not `x` but `data`. Maybe also add a more descriptive message, eg.:
```suggestion
        raise ValueError("Data has incompatible timeslicing with target timeslice during broadcast operation.")
```
Or something along these lines.